substrate/frame/contracts/src/debug.rs

```diff
@@ -70,3 +74,37 @@ impl CallSpan for () {
 		log::trace!(target: LOG_TARGET, "call result {output:?}")
 	}
 }
+
+/// Provides an interface for intercepting contract calls.
+pub trait CallInterceptor<T: Config> {
+	/// Allows to intercept contract calls and decide whether they should be executed or not.
+	/// If the call is intercepted, the mocked result of the call is returned.
+	///
+	/// # Arguments
+	///
+	/// * `contract_address` - The address of the contract that is about to be executed.
+	/// * `entry_point` - Describes whether the call is the constructor or a regular call.
+	/// * `input_data` - The raw input data of the call.
+	///
+	/// # Expected behavior
+	///
+	/// This method should return:
+	/// * `Some(ExecResult)` - if the call should be intercepted and the mocked result of the call
+	/// is returned.
+	/// * `None` - otherwise, i.e. the call should be executed normally
+	fn intercept_call(
+		contract_address: &T::AccountId,
+		entry_point: ExportedFunction,
```

Comment:
```suggestion
		entry_point: &ExportedFunction,
```

Reply:
`ExportedFunction` implements `Copy` and is very lightweight, thus no need for passing a reference; however, if you insist I can alter the API

Reply:
well it's already done if you accept suggested changes ;P

Reply:
:white_check_mark: